services/visual_recognition/v3.js

```diff
@@ -0,0 +1,348 @@
+            message ='Node has been unable to open the image stream'; 
+            node.error(message, msg);
+            return;        
+          }  
+          stream_buffer(info.path, msg.payload, function () {
+          params['images_file'] = fs.createReadStream(info.path);
+          if ( msg.params != null && msg.params.classifier_ids != null)
+            params['classifier_ids']=msg.params['classifier_ids'];
+          if ( msg.params != null && msg.params.owners != null)
+            params['owners']=msg.params['owners'];
+          if ( msg.params != null && msg.params.threshold != null)
+            params['threshold']=msg.params['threshold'];
+          performAction(params, feature, actionComplete);
+          });
+        }); // temp
+      } else if (feature=='createClassifier') {   
+          var list_params = {};
+          var asyncTasks = [];
+          var prop = null;
+          for (var k in msg.params)
+          {
+            prop = k;
+            if (prop.indexOf('_examples')>=0)
+            {
+              // before pushing the function into the task array wrap the push in an IIFE function, passing in the 'prop' parameter
+              (function(prop, list_params, msg) {
+
+               asyncTasks.push(function (callback) {
+                  var buffer = msg.params[prop];
+                  temp.open({suffix: '.' + fileType(buffer).ext}, function (err, info) {
+                    if (err) {
+                      this.status({fill:'red', shape:'ring', text:'unable to open image stream'});          
+                      var message ='Node has been unable to open the image stream'; 
+                      node.error(message, msg);
+                      callback('open error on '+prop);
+                    }  
+                    stream_buffer(info.path, msg.params[prop], function () {
+                      list_params[prop]=fs.createReadStream(info.path);
+                      callback(null, prop);
+                    });
+                  }); // temp.open
+              }); // asyncTasks.push
+
+              })(prop, list_params, msg);
+
+            } else if (prop=='name') {
+              list_params[prop]=msg.params[prop];
+            }
+          } // for
+          
+
+          async.parallel(asyncTasks, function(error, results){
+            // when all temp local copies are ready, copy of all parameters and request to watson api
+            console.log(error, results);
+            if (error)
+            {
+              console.log("Parallel ended with error " + error);
+              return;
+            }
+            for (p in list_params)
+              params[p]=list_params[p];
+            performAction(params, feature, actionComplete2);
+          });
+      }
+        else if (feature=='retrieveClassifiersList') { 
+          performAction(params, feature, actionComplete2);
+      }
+        else if (feature =='retrieveClassifierDetails') {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected '===' and instead saw '=='.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2686950655/issues/source?bid=3225115&fileBranchId=3338101#l309)